lib/chai/core/assertions.js

```diff
@@ -1351,11 +1359,13 @@ module.exports = function (chai, _) {
 
   function assertKeys (keys) {
     var obj = flag(this, 'object')
+      , isDeep = flag(this, 'deep')
       , str
       , ok = true
       , mixedArgsMsg = 'when testing keys against an object or an array you must give a single Array|Object|String argument or multiple String arguments';
 
     if (_.type(obj) === 'map' || _.type(obj) === 'set') {
+      str = isDeep ? 'deeply ' : '';
```

Comment:
I realize messages aren't being tested yet due to an issue with inspect and sets/maps, but just noting that I believe `str` will get overwritten [here](https://github.com/chaijs/chai/blob/01963b74389f35a6f13739383015ac945987485c/lib/chai/core/assertions.js#L1430-L1436).